service/history/task/transfer_task_executor_base.go

```diff
@@ -458,7 +458,14 @@ func appendContextHeaderToSearchAttributes(attr, context map[string][]byte, allo
 		}
 		// context header are raw string bytes, need to be json encoded to be stored in search attributes
 		// ignore error as it can't happen to err on json encoding string
-		data, _ := json.Marshal(string(v))
+		// context header can contain sensitive information, so we need to hide the value if it is in the hiddenValueKeys
+		var val string
+		if _, ok := hiddenValueKeys[key]; ok {
```

Comment:
Since you made it a map with bool value, it might be confusing if someone sets the value to false but still see it getting redacted. so let's check the value.
Also check if `hiddenValueKeys` is nil or not.

Reply:
Yeah, moved to a helper function, added checks there